pkg/analysis/passes/backenddebug/backenddebug.go

```diff
@@ -28,35 +35,49 @@ func run(pass *analysis.Pass) (interface{}, error) {
 	if !ok {
 		return nil, nil
 	}
-	metadataBody, ok := pass.ResultOf[metadata.Analyzer].([]byte)
+
+	metadatamap, ok := pass.ResultOf[nestedmetadata.Analyzer].(nestedmetadata.Metadatamap)
 	if !ok {
 		return nil, nil
 	}
 
-	var data metadata.Metadata
-	if err := json.Unmarshal(metadataBody, &data); err != nil {
-		return nil, err
+	hasExecutable := false
+	for _, data := range metadatamap {
+		if data.Executable != "" {
+			hasExecutable = true
+			break
+		}
 	}
-	if data.Executable == "" {
-		// Do not perform check for plugins without a backend executable
+
+	// don't evaluate for plugins that don't have an executable
+	if !hasExecutable {
 		return nil, nil
 	}
 
-	for _, fn := range []string{"standalone.txt", "pid.txt"} {
-		if _, err := os.Stat(filepath.Join(archiveDir, fn)); err != nil {
-			if os.IsNotExist(err) {
-				// Banned file not found
-				continue
-			}
-			return nil, fmt.Errorf("stat %q: %w", fn, err)
-		}
-		// Found a banned file
+	textFiles, err := doublestar.FilepathGlob(archiveDir + "/**/*.txt")
```

Comment:
changed the logic to search for the txt files across all plugins inside an archive.